packages/flutter_tools/lib/src/android/README.md

```diff
@@ -0,0 +1,90 @@
+# Flutter Tools for Android
+
+This section of the Flutter repository contains the command line developer tools
+for building Flutter applications on Android. What follows are some notes about
+updating this part of the tool.
+
+## Updating Android dependencies
+The Android dependencies that may need updating in Flutter project templates
+include the Android NDK and SDK versions, Gradle, the Kotlin Gradle Plugin,
+and the Android Gradle Plugin. The current template versions of these
+dependencies can be found in [gradle_utils.dart](gradle_utils.dart). The
+following details steps to take when (i) wanting to upgrade the template
+versions of these dependencies in `gradle_utils.dart` and (ii) when new
+versions of these dependencies become available to ensure that the tool
+is up-to-date. **Please make sure to check for new available versions of
+the dependency whose version you are upgrading!**
+
+### Android NDK & SDK versions
+All of the Android SDK/NDK versions noted in `gradle_utils.dart`
+(`compileSdkVersion`, `minSdkVersion`, `targetSdkVersion`, `ndkVersion`) versions should match the values in Flutter Gradle Plugin (FlutterExtension), so updating any of
+these versions also requires an update in [flutter.groovy](../../../gradle/src/main/groovy/flutter.groovy).
+
+The Flutter Gradle plugin is only applied to app projects, and modules that are built from source
+using (`include_flutter.groovy`). The remaining projects are: plugins, and modules compiled as AARs. In modules, the ephemeral directory `.android` is always regenerated after `flutter pub get`, so new versions are picked up after a Flutter upgrade.
+
+When updating the Android `compileSdkVersion`, `minSdkVersion`, or
+`targetSdkVersion`, make sure that:
+- Framework integration & benchmark tests are running with at least that SDK
+version.
+- Flutter tools tests that perform String checks with the current template
+SDK verisons are updated (you should see these fail if you do not fix them
+preemptively).
```

Comment:
After talking with @christopherfujino, I will file an issue and plan to work on refactoring most of our references to these various versions in the hopes of lowering the amount of steps required to accommodate updating templates or acknowledging new version dependencies.

For instance, many of our tests can be refactored in terms of our constants such that we would only need to add new test cases when new versions come out versus having to do that in addition to modifying pre-existing test cases.

Reply:
Can you include a link to that bug you filed here in a comment and/or in the readme?